ui/src/app/components/object-detail/object-detail.component.ts

```diff
@@ -1263,22 +1264,25 @@ export class ObjectDetailComponent implements OnInit {
     this.localIndexData.forEach((idx) => {
       if (idx.spColName) spIndexCount += 1
     })
-    this.localIndexData.push({
-      spColName: this.addIndexKeyForm.value.columnName,
-      spDesc: this.addIndexKeyForm.value.ascOrDesc === 'desc',
-      spOrder: spIndexCount + 1,
-      srcColName: '',
-      srcDesc: undefined,
-      srcOrder: '',
-      srcColId: undefined,
-      spColId: this.currentObject
-        ? this.conversion.getColIdFromSpannerColName(
-          this.addIndexKeyForm.value.columnName,
-          this.currentObject.parentId,
-          this.conv
-        )
-        : '',
-    })
+    if (typeof this.addIndexKeyForm.value.columnName === "string") {
```

Comment:
I think we are having to do this because the time of defining the form we did not define the datatype? Is there a way to define the data type at the time of form declaration?

```
addIndexKeyForm = new FormGroup({
    columnName: new FormControl('', [Validators.required]),
    ascOrDesc: new FormControl('', [Validators.required]),
  })
```

I think you need to do this - 
https://angular.io/guide/typed-forms#specifying-an-explicit-type

This should take care of all the `if` checks added in the code.

Reply:
Removed the if check and added ! - non null assertion operator